C++/include/utils/data_validation.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+    Data validation
+    ---------------
+    This utility function contains two sub-funtions :
+    
+    1) make_upper_case() : converts the characters in a string to upper case .
+    
+    2) make_lower_case() : converts the characters in a string to lower case . 
+
+    Time complexity
+    ---------------
+    O(n), where n is the length of the string .
+*/
+
+#ifndef DATA_VALIDATION_HPP
+#define DATA_VALIDATION_HPP
+
+#include <string>
+
+using std::string;
+
+/*
+    make_upper_case()
+    -----------------
+    This function returns a given string into upper case . 
+    Loop invariant : Selects each character and converts it to upper case .
+
+    Return value
+    ------------
+    string having all the characters in upper case .
+*/
+
+string make_upper_case(string str)
+{
+    //converting each character to upper case using for loop
+    for (auto &c : str)
+    {
+        c = std::toupper(c); // using the function toupper() for conversion
```

Comment:
Rather than just using `std::toupper()` and `std::tolower()`, I think it would be better to implement this manually (as done [here](https://github.com/mrKappen/Algos/blob/662de4b9a254fe5da228c93334549b7fcf9e2f7d/C%2B%2B/include/utils/data_validation.hpp), though there are lots of ways to do it). Obviously in "real life" you'd just use the `std` function, but the point of this repo is (generally) to implement things ourselves (i.e., sorting a list manually rather than just calling `std::sort()`). More good info [here](https://stackoverflow.com/questions/22425825/changing-a-lowercase-character-to-uppercase-in-c).

So, please implement these utilities without using the `std` functions. Let me know if you have any questions!

Reply:
Okay...I will dive right into it !